xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp

```diff
@@ -8,13 +8,20 @@
 
 #include "SavestateDatabase.h"
 #include "SavestateFlatBuffer.h"
-#include "SavestateUtils.h"
 #include "URL.h"
+#include "FileItem.h" 
+#include "filesystem/Directory.h"
 #include "filesystem/File.h"
 #include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "XBDateTime.h"
+
+constexpr auto SAVESTATE_EXTENSION = ".sav";
+constexpr auto SAVESTATE_BASE_FOLDER = "special://home/saves/";
 
 using namespace KODI;
 using namespace RETRO;
+using namespace XFILE;
```

Comment:
As you know, the `using` keyword imports all known symbols from the namespace into the global namespace. For huge projects like Kodi, if we were to do this for every component, it would lead to explosive growth of global namespace population. Therefore, try to avoid importing unrelated namespaces outside of the ones this class is declared within.

Reply:
Also, can you add `#include "filesystem/IFileTypes.h"` to this file? It's needed for `XFILE::READ_TRUNCATED`, which doesn't seem to be introduced by your PR, but additional lines added don't hurt a PR that is mostly lines added.